File: 1.cuenta-bancaria/main.cs
using System;

public class CuentaBancaria
{
    // Atributo privado (Encapsulamiento)
    private decimal _saldo;

    public CuentaBancaria(decimal saldoInicial = 0)
    {
        _saldo = saldoInicial >= 0 ? saldoInicial : 0;
    }

    public void Depositar(decimal monto)
    {
        if (monto > 0)
        {
            _saldo += monto;
            Console.WriteLine($"Depósito exitoso: +${monto:F2}");
        }
        else
        {
            Console.WriteLine("El monto a depositar debe ser mayor a 0.");
        }
    }

    public void Retirar(decimal monto)
    {
        if (monto > 0 && monto <= _saldo)
        {
            _saldo -= monto;
            Console.WriteLine($"Retiro exitoso: -${monto:F2}");
        }
        else
        {
            Console.WriteLine("Retiro rechazado: Fondos insuficientes o monto inválido.");
        }
    }

    public decimal ObtenerSaldo()
    {
        return _saldo;
    }
}

class Program
{
    static void Main()
    {
        CuentaBancaria miCuenta = new CuentaBancaria(100m);
        miCuenta.Depositar(50m);
        miCuenta.Retirar(30m);
        miCuenta.Retirar(200m);
        Console.WriteLine($"Saldo final: ${miCuenta.ObtenerSaldo():F2}");
    }
}

File: 1.cuenta-bancaria/main.cpp
#include <iostream>
#include <iomanip>

class CuentaBancaria {
private:
    // Atributo privado (Encapsulamiento)
    double saldo;

public:
    CuentaBancaria(double saldoInicial = 0.0) {
        saldo = (saldoInicial >= 0) ? saldoInicial : 0.0;
    }

    void depositar(double monto) {
        if (monto > 0) {
            saldo += monto;
            std::cout << "Deposito exitoso: +$" << std::fixed << std::setprecision(2) << monto << "\n";
        } else {
            std::cout << "El monto a depositar debe ser mayor a 0.\n";
        }
    }

    void retirar(double monto) {
        if (monto > 0 && monto <= saldo) {
            saldo -= monto;
            std::cout << "Retiro exitoso: -$" << std::fixed << std::setprecision(2) << monto << "\n";
        } else {
            std::cout << "Retiro rechazado: Fondos insuficientes o monto invalido.\n";
        }
    }

    double obtenerSaldo() const {
        return saldo;
    }
};

int main() {
    CuentaBancaria miCuenta(100.0);
    miCuenta.depositar(50.0);
    miCuenta.retirar(30.0);
    miCuenta.retirar(200.0);
    std::cout << "Saldo final: $" << std::fixed << std::setprecision(2) << miCuenta.obtenerSaldo() << "\n";

    return 0;
}